<?xml version="1.0"?>
<!--
kinects_visualization.launch: Rviz visualization
Author: Ravi Joshi
Date: 2017/24/10
-->
<launch>
  <arg name="base_frame_id" default="base"/>
  <arg name="pc1_frame_id" default="kinect1_link"/>
  <arg name="pc2_frame_id" default="kinect2_link"/>
  <arg name="pc3_frame_id" default="kinect3_link"/>
  <arg name="tf_wait" default="2.0"/>
  <!--
  colors in hex for each kinect. make sure to have
  only one space in between each color as it is below
  #212223 gray black
  -->
  <arg name="colors" default="[#212223, #212223, #212223]"/>
  <arg name="scale" default="1.0"/>
  <arg name="freq" default="0.5"/>

  <node name="kinects_visualization" pkg="multiple_kinect_baxter_calibration" type="kinects_visualization.py" output="screen">
    <param name="base_frame_id" value="$(arg base_frame_id)"/>
    <param name="pc1_frame_id" value="$(arg pc1_frame_id)"/>
    <param name="pc2_frame_id" value="$(arg pc2_frame_id)"/>
    <param name="pc3_frame_id" value="$(arg pc3_frame_id)"/>
    <param name="tf_wait" value="$(arg tf_wait)"/>
    <param name="colors" value="$(arg colors)"/>
    <param name="scale" value="$(arg scale)"/>
    <param name="freq" value="$(arg freq)"/>
  </node>
</launch>
